patches/packages/luci/0005-model.uci-add-add_to_set-remove_from_set.patch

```diff
@@ -0,0 +1,90 @@
+From: Nils Schneider <nils@nilsschneider.net>
+Date: Mon, 17 Aug 2015 20:39:58 +0200
+Subject: model.uci: add add_to_set / remove_from_set
+
+Adds two functions to simplify working with UCI lists:
+
+- add_to_set, which ensures a given value will be present in a list, and
+- remove_from_set, which removes a value from list.
+
+I've called these methods "set" because they treat the list as a set,
+i.e. duplicated values will be removed. Also, order is not preserved.
+
+Signed-off-by: Nils Schneider <nils@nilsschneider.net>
+
+diff --git a/modules/luci-base/luasrc/model/uci.lua b/modules/luci-base/luasrc/model/uci.lua
+index 1659137..74ecac6 100644
+--- a/modules/luci-base/luasrc/model/uci.lua
++++ b/modules/luci-base/luasrc/model/uci.lua
+@@ -147,6 +147,36 @@ function Cursor.set_list(self, config, section, option, value)
+ 	return false
+ end
+ 
++function Cursor.add_to_set(self, config, section, option, value, remove)
++	local list = self:get_list(config, section, option)
++
++	if not list then
++		return false
++	end
++
++	local set = {}
++	for _, l in ipairs(list) do
++		set[l] = true
++	end
++
++	if remove then
++		set[value] = nil
++	else
++		set[value] = true
++	end
++
++	list = {}
++	for k, _ in pairs(set) do
++		table.insert(list, k)
++	end
++
++	return self:set_list(config, section, option, list)
```

Comment:
I've noticed that this breaks when you try to remove the last element of a set (as the `set` and `set_list` function of UCI don't allow setting an empty table). I'm not sure if this should be fixed here or changed in uci itself. Also, `set_list` is a weird wrapper around `set`, just using `set` is probably preferable.